File: mrmini.jl
import Random
import TimeSeriesClassification

# Precompile with small calculation
r = Random.rand(Float32, 10, 100)
dilations, num_features_per_dilation, biases = TimeSeriesClassification.MiniRocket._MiniRocket.fit(r)
out = TimeSeriesClassification.MiniRocket._MiniRocket.transform(r, dilations=dilations, num_features_per_dilation=num_features_per_dilation, biases=biases)


# Run 1
# 8c, 502.089 ms
# 1c, 918.479 ms
r = Random.rand(Float32, 100, 10000)

@time begin
    dilations, num_features_per_dilation, biases = TimeSeriesClassification.MiniRocket._MiniRocket.fit(r)
    TimeSeriesClassification.MiniRocket._MiniRocket.transform(r, dilations=dilations, num_features_per_dilation=num_features_per_dilation, biases=biases)
end


# Run 2
# 8c, 4940.528 ms
# 1c, 9100.596 ms
r = Random.rand(Float32, 100, 100000)

@time begin
    dilations, num_features_per_dilation, biases = TimeSeriesClassification.MiniRocket._MiniRocket.fit(r)
    TimeSeriesClassification.MiniRocket._MiniRocket.transform(r, dilations=dilations, num_features_per_dilation=num_features_per_dilation, biases=biases)
end


# Run 3
# 8c, 57710.066 ms, 37238 MB RAM
# 1c, 92917.752 ms, 37238 MB RAM
r = Random.rand(Float32, 100, 1000000)

@time begin
    dilations, num_features_per_dilation, biases = TimeSeriesClassification.MiniRocket._MiniRocket.fit(r)
    TimeSeriesClassification.MiniRocket._MiniRocket.transform(r, dilations=dilations, num_features_per_dilation=num_features_per_dilation, biases=biases)
end

File: mrmini.py
from sktime.transformations.panel.rocket._minirocket_numba import _fit, _transform
import time
import numpy as np
import numba

numba.set_num_threads(8)

# Precompile
data = np.random.rand(10, 100).astype(np.float32)
mini = _fit(data)
_transform(data, mini)

# Run 1
# 8c, 857.288122177124 ms
data = np.random.rand(100, 10000).astype(np.float32)
start = time.time()
mini = _fit(data)
_transform(data, mini)
end = time.time()
print(end - start)

# Run 2
# 8c, 13082.681655883789 ms
data = np.random.rand(100, 100000).astype(np.float32)
start = time.time()
mini = _fit(data)
_transform(data, mini)
end = time.time()
print(end - start)

# Run 3
# 8c, 314905.8892726898 ms
data = np.random.rand(100, 1000000).astype(np.float32)
start = time.time()
mini = _fit(data)
_transform(data, mini)
end = time.time()
print(end - start)

